### PR TITLE
[Merged by Bors] - chore: Golf `Finset.exists_subsuperset_card_eq`

### PR DESCRIPTION
**Pull request by YaelDillies**
Thanks Bhavik!

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Finset/Card.lean [hunk 1]
```diff
@@ -623,15 +623,11 @@ set `u` of size `n` which is both a superset of `s` and a subset of `t`. -/
...
-  obtain ⟨k, hk⟩ := Nat.le.dest hnt
-  clear hnt
-  induction' k with k ih generalizing t
-  · exact ⟨t, hst, Subset.rfl, hk.symm⟩
-  obtain ⟨a, ha⟩ : (t \ s).Nonempty := by rw [← card_pos, card_sdiff hst]; omega
-  replace hst : s ⊆ t.erase a := subset_erase.2 ⟨hst, (mem_sdiff.1 ha).2⟩
-  replace hk : n + k = card (erase t a) := by rw [card_erase_of_mem (mem_sdiff.1 ha).1]; omega
-  obtain ⟨u, hsu, hut, hu⟩ := ih hst hk
-  exact ⟨u, hsu, hut.trans (erase_subset ..), hu⟩
+  refine Nat.decreasingInduction' ?_ hnt ⟨t, by simp [hst]⟩
```
Comment (eric-wieser) [inline Mathlib/Data/Finset/Card.lean:626]: This bugged me, so I've made a start on https://github.com/leanprover-community/mathlib4/pull/14431

Comment (YaelDillies) [reply Mathlib/Data/Finset/Card.lean:626]: Nerdsniping Eric is the best kind of nerdsnipe 😝

**Review by rwst — APPROVED** [review]
LGTM

File: Mathlib/Data/Finset/Card.lean (hunk 1 above)
Comment (b-mehta) [reply Mathlib/Data/Finset/Card.lean:626]: This was the best possible outcome of this golf :D

**Review by b-mehta — APPROVED** [review]
lgtm but I wrote it, so someone else should review too.

**Comment by eric-wieser** [comment]
I'd prefer to block this on #14431, to motivate review there once I finish up in an hour or so

**Review by eric-wieser — APPROVED** [review]
Nevermind, I can't actually make `decreasingInduction'` work with `induction` after all, it's not an induction principle in the usual sense because the inductive step depends on the original value.

bors merge

**Pull request closed without merging**